packages/application-shell/src/components/custom-view-context/custom-view-context.tsx

```diff
@@ -0,0 +1,68 @@
+import { createContext, type ReactNode, useContext, useMemo } from 'react';
+
+/*
+  TODO: These types are temporary until we have the proper
+  ones generated from the Settings schema
+*/
+export type TPermissionGroup = {
+  name: string;
+  oAuthScopes: string[];
+};
+
+export type TCustomView = {
+  id: string;
+  defaultLabel: string;
+  labelAllLocales: Record<string, string>;
+  url: string;
+  type: 'CustomPanel' | 'CustomTab';
+  typeConfig?: {
+    size?: 'SMALL' | 'LARGE';
+  };
+  locators: string[];
+  permissions: TPermissionGroup[];
+};
+
+type TCustomViewContext = {
+  config: TCustomView;
+};
+
+type TCustomViewContextProviderProps = {
+  config: TCustomView;
+  children: ReactNode;
+};
+
+const CustomViewContext = createContext<TCustomViewContext>({
```

Comment:
This is useful to have the CustomView configuration easily available in its components tree.